source/gameengine/Ketsji/KX_FontObject.cpp

```diff
@@ -113,6 +117,85 @@ void KX_FontObject::ProcessReplica()
 	KX_GameObject::ProcessReplica();
 }
 
+void KX_FontObject::AddMeshUser()
+{
+	m_meshUser = new RAS_TextUser(m_pClient_info);
+
+	// set the part of the mesh slot that never change
+	float *fl = GetOpenGLMatrixPtr()->getPointer();
+	m_meshUser->SetMatrix(fl);
+
+	RAS_BucketManager *bucketManager = GetScene()->GetBucketManager();
+	bool created = false;
+	RAS_MaterialBucket *bucket = bucketManager->FindBucket(GetTextMaterial(), created);
+
+	// If the material bucket is just created then we add a new mesh slot.
+	if (created) {
+		bucket->AddMesh(NULL);
+	}
+
+	/* We copy the original mesh slot which is at the begin of the list, if it's not the case it
+	 * doesn't matter as the mesh slot are all similar exepted their mesh user pointer which is
+	 * set to NULL in copy. By copying instead of adding a mesh slot we reuse the same display
+	 * array bucket.
+	 */
+	RAS_MeshSlot *ms = bucket->CopyMesh(*bucket->msBegin());
+	ms->SetMeshUser(m_meshUser);
+	ms->SetDeformer(NULL);
+	m_meshUser->AddMeshSlot(ms);
+}
+
+void KX_FontObject::UpdateBuckets()
+{
+	// Update datas and add mesh slot to be rendered only if the object is not culled.
+	if (m_bVisible && m_meshUser) {
+		if (m_pSGNode->IsDirty()) {
+			GetOpenGLMatrix();
+		}
+
+		// Allow for some logic brick control
+		if (GetProperty("Text")) {
+			m_text = split_string(GetProperty("Text")->GetText());
+		}
+
+		// update the animated color
+		GetObjectColor().getValue(m_color);
```

Comment:
The patch is not finished? A normal material will be added in blender data conversion ConvertMesh to have diffuse, specular etc? Like for other meshes? Or is it impossible?